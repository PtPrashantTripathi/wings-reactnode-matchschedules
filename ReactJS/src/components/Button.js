// Button component
// props:
//   onClick - a function calls native onClick event
//   children - renders as child
//   className - helps to style the component
const Button = ({ onClick, children, className }) => (
  <button onClick={onClick} className={className}>
    {children}
  </button>
);

export default Button;