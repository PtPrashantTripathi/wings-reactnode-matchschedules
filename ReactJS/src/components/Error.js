// Component that renders error message.
// props:
//   message: error message that has to be displayed
const Error = ({ message }) => (
  message ? <div className="error">{message}</div> : null
);


export default Error;