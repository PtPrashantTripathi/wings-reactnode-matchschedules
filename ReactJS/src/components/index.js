export { default as List } from './List';
export { default as Card } from './Card';
export { default as Modal } from './Modal';
export { default as Button } from './Button';
export { default as Error } from './Error';
export { default as Form } from './Form';