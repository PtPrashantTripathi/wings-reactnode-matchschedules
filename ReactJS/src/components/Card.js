import Button from "./Button";

// Card component renders each record in "data.records" from the api
// consists of table, thead, tbody elements along with tr, th, td
// props:
//   deleteHandler: a function that deletes the record
//   _id: _id from the api 
//   index: each row index 
//   count: count from the api
//   venue: venue key from the api
//   team1: team1 key from the api
//   team2: team2 key from the api
//   date: date key from the api. Make sure to render it in format "Day Mon DD YYYY" ex: "Sun Jun 23 2021"

const Card = ({
  _id,
  index,
  count,
  venue,
  team1,
  team2,
  date
}) => (
    <div className="card" id={_id}>
      <div className="card-header">
        <span>
          <span className="small-txt">Match </span>
          {index}
          <span className="small-txt"> of </span>
          {count}
        </span>
        <span>
          <span className="small-txt">Venue: </span>
          {venue}
        </span>
      </div>
      <span id="teams">
        {team1}
        <span className="small-txt"> vs </span>
        {team2}
      </span>
      <div className="card-footer">
        <span id="matchDate" className="small-txt">
          {/* Render the date here in format "Day Mon DD YYYY" (ex: "Sun Jun 23 2021") */}
          {new Date(date).toDateString()}
        </span>
      </div>
    </div>
  );

export default Card;