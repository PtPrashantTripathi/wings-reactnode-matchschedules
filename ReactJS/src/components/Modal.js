import Button from "./Button";

// Renders a Modal
// props:
//   show: prop willÂ can be used to show/hide the modal
//   closeHandler: prop is used by Button, will close the modal
const Modal = ({ children, show, closeHandler }) => (
  <>
    {show ?
      <div className="modal-bg">
        <div className="modal">
          <Button onClick={closeHandler} className="btn-close">&#x2715;</Button>
          {children}
        </div>
      </div> : null
    }
  </>
);

export default Modal;