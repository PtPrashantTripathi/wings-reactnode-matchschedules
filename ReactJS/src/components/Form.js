import model from '../model';
import Button from "./Button";
import Error from "./Error";

// Form Component has following elements
// use label and input elements for each input field
// custom Button component executes "onSaveHandler" prop when clicked
// props:
//   title: renders heading using h3 tag
//   onSaveHandler: called when the Save button is clicked
//   inputOnChangeHandler: helps to capture the entered data to store it in state
//   error: renders the error from backend
const Form = ({ title, inputOnChangeHandler, onSaveHandler, error }) => (
  <form>
    <h3>{title}</h3>
    <Error message={error} />
    {model.filter(i => i.input_type)
      .map(i =>
        <div key={i.key} className="form-group">
          <label htmlFor={i.column_name}>{i.title}</label>
          <input
            type={i.input_type}
            name={i.column_name}
          />
        </div>
      )}
    <Button onClick={onSaveHandler} className="success">Save</Button>
  </form>
);

export default Form;