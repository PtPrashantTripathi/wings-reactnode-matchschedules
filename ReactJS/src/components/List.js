import Card from "./Card";


// props:
//   data: consists of data records to be rendered
const List = ({ data }) => (
  data ? (<div className="list">
    {/* Your code goes here */}
    {/* Render the Card with required props here */}
  </div>) : null
);


export default List;