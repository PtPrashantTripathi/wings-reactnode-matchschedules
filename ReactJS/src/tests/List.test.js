import { create } from 'react-test-renderer';
import { List, Button, Card } from '../components';
import testData from './testData';

describe('List', () => {
  test('renders List component', () => {
    const component = create(<List data={testData} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders List with data', () => {
    const deleteHandler = jest.fn();
    const component = create(<List data={testData} />);
    const instance = component.root;
    const cards = instance.findAllByType(Card);
    const { count, records } = testData;
    
    expect(instance.props.data).toEqual(testData);
    expect(cards.length).toBe(4);

    cards.forEach((card, i) => {
      expect(card.props.index).toBe(i + 1);
      expect(card.props.count).toEqual(count);
      expect(card.props._id).toEqual(records[i]['_id']);
      expect(card.props.venue).toEqual(records[i]['venue']);
      expect(card.props.team1).toEqual(records[i]['team1']);
      expect(card.props.team2).toEqual(records[i]['team2']);
      expect(card.props.date).toEqual(records[i]['date']);
    });
  });

});