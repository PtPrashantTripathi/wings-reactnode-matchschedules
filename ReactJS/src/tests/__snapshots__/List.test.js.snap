// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders List component 1`] = `
<div
  className="list"
>
  <div
    className="card"
    id="5f5b0c42ccf6cf072812d951"
  >
    <div
      className="card-header"
    >
      <span>
        <span
          className="small-txt"
        >
          Match 
        </span>
        1
        <span
          className="small-txt"
        >
           of 
        </span>
        4
      </span>
      <span>
        <span
          className="small-txt"
        >
          Venue: 
        </span>
        Delhi
      </span>
    </div>
    <span
      id="teams"
    >
      CSK
      <span
        className="small-txt"
      >
         vs 
      </span>
      MI
    </span>
    <div
      className="card-footer"
    >
      <span
        className="small-txt"
        id="matchDate"
      >
        Sun May 23 2021
      </span>
    </div>
  </div>
  <div
    className="card"
    id="5f5b0c42ccf6cf072812d952"
  >
    <div
      className="card-header"
    >
      <span>
        <span
          className="small-txt"
        >
          Match 
        </span>
        2
        <span
          className="small-txt"
        >
           of 
        </span>
        4
      </span>
      <span>
        <span
          className="small-txt"
        >
          Venue: 
        </span>
        Mumbai
      </span>
    </div>
    <span
      id="teams"
    >
      RCB
      <span
        className="small-txt"
      >
         vs 
      </span>
      SRH
    </span>
    <div
      className="card-footer"
    >
      <span
        className="small-txt"
        id="matchDate"
      >
        Mon May 24 2021
      </span>
    </div>
  </div>
  <div
    className="card"
    id="5f5b0c42ccf6cf072812d953"
  >
    <div
      className="card-header"
    >
      <span>
        <span
          className="small-txt"
        >
          Match 
        </span>
        3
        <span
          className="small-txt"
        >
           of 
        </span>
        4
      </span>
      <span>
        <span
          className="small-txt"
        >
          Venue: 
        </span>
        Chennai
      </span>
    </div>
    <span
      id="teams"
    >
      PBKS
      <span
        className="small-txt"
      >
         vs 
      </span>
      SRH
    </span>
    <div
      className="card-footer"
    >
      <span
        className="small-txt"
        id="matchDate"
      >
        Wed May 26 2021
      </span>
    </div>
  </div>
  <div
    className="card"
    id="5f5b0c42ccf6cf072812d954"
  >
    <div
      className="card-header"
    >
      <span>
        <span
          className="small-txt"
        >
          Match 
        </span>
        4
        <span
          className="small-txt"
        >
           of 
        </span>
        4
      </span>
      <span>
        <span
          className="small-txt"
        >
          Venue: 
        </span>
        Bengaluru
      </span>
    </div>
    <span
      id="teams"
    >
      RR
      <span
        className="small-txt"
      >
         vs 
      </span>
      DC
    </span>
    <div
      className="card-footer"
    >
      <span
        className="small-txt"
        id="matchDate"
      >
        Fri May 28 2021
      </span>
    </div>
  </div>
</div>
`;
